frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { ConfigMapModel, ProjectRequestModel, ProjectModel } from '@console/internal/models';
+import { K8sResourceKind, k8sGet, k8sCreate, k8sPatch } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+
+// can't create project with name prefix with 'openshift-*', once we have proxy need to update
+const USER_SETTING_CONFIGMAP_NAMESPACE = 'console-user-settings';
+
+export const useUserSettings = (key: string, defaultValue: string = '') => {
+  const userUid = useSelector(
+    (state: RootState) => state.UI.get('user')?.metadata?.uid ?? 'kubeadmin',
+  );
+
+  const [configMapResource, configMapData] = React.useMemo(
+    () => [
+      {
+        kind: ConfigMapModel.kind,
+        namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+        isList: false,
+        name: `user-settings-${userUid}`,
+      },
+      {
+        apiVersion: ConfigMapModel.apiVersion,
+        kind: ConfigMapModel.kind,
+        metadata: {
+          name: `user-settings-${userUid}`,
+          namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+        },
+        data: {
+          [key]: defaultValue,
+        },
+      },
+    ],
+    [userUid, key, defaultValue],
+  );
+
+  const [cfData, cfLoaded, cfLoadError] = useK8sWatchResource<K8sResourceKind>(configMapResource);
+  const [settings, setSettings] = React.useState('');
```

Comment:
Should this be undefined?
It cannot be the default value because we don't want a flash of the default value while we're still loading the settings.